src/NuGet.Core/NuGet.Common/NuGet.Common.csproj

```diff
@@ -25,6 +25,10 @@
     <ProjectReference Include="..\NuGet.Frameworks\NuGet.Frameworks.csproj" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+    <PackageReference Include="Microsoft.IO.Redist" />
```

Comment:
The code looks good but adding this dependency means we'll need to ILMerge it into `NuGet.exe` and ship it in the NuGet VSIX.

Reply:
Is there a binding redirect for this in VS? 
If no, then once we add, we might get an RPS failure too.

Given that we're already dealing with some RPS issues, it'd be great to hold on making this change until that happens.

Reply:
CPS uses this. I believe you should be fine. Perhaps @arkalyanms has some context here.

Reply:
Even if VS has a binding redirect/codebase for the dll, msbuild cli restore doesn't use any VS infrastucture. I see that msbuild's bin directory has the dll, so maybe NuGet doesn't need it, although if msbuild and nuget ever compile against different versions of the dll, I expect there will be errors unless nuget ships its own copy in its own directory.

So, I'm not sure it's "safe" to make this change, without changing nuget's vsix to include microsoft.io.redist.dll.

Reply:
This assembly ships in C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\PublicAssemblies and should not be included in NuGet's VSIX.

Reply:
MSBuild doesn't load assemblies from VS's PublicAssemblies, so anything used by our MSBuild tasks have to be included in NuGet's VSIX, or command  line (all CI) restores will fail. If the assembly ships in MSBuild's own bin directory, that might be an exception to the rule, but it needs to be tested.

Reply:
NuGet is an extension to MSBuild and VS. It should be inheriting dependencies of the dll in both cases. The fact that VS and MSBuild bits are mixed in a single location for NuGet is unusual and not how other features work, so the trickiness here is to pick a dependency that satisifies both VS and MSBuild.

Reply:
It took me longer than I expected it would, but msbuild restore still works with this change. And nuget.exe's ilmerge uses the bin directory's `*.dll`, so it'll get included too.

So, as far as I can tell, potential RPS/DDRITs regressions excluded, this PR probably won't break anything.